taichi/ir/frontend_ir.h

```diff
@@ -464,47 +464,26 @@ class GlobalPtrExpression : public Expression {
   }
 };
 
-class GlobalTensorElementExpression : public Expression {
+class TensorElementExpression : public Expression {
  public:
   Expr var;
   ExprGroup indices;
-  int cols{0};
-  bool is_aos{false};
+  ExprGroup shape;
```

Comment:
nit: We can just use `std::vector<int>` here. The const-ness is encoded in the type itself, (instead of us having to assert the contents are ConstStmt)